src/frontend/src/handler/create_index.rs

```diff
@@ -327,6 +331,7 @@ fn assemble_materialize(
         table_name,
         ScanTableType::default(),
         table_desc.clone(),
+        Some(table_catalog),
```

Comment:
We have 3 different table types here we maintain. It should be merged into a single `TableType`.